app/components/profile_picture_button.js

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+import {Client4} from 'mattermost-redux/client';
+
+import {t} from 'app/utils/i18n';
+
+import AttachmentButton from 'app/components/attachment_button';
+
+export default class ProfilePictureButton extends PureComponent {
+    static propTypes = {
+        currentUser: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        extraOptionsAction: PropTypes.func,
+    };
+
+    state = {
+        extraOptions: null,
+    };
+
+    constructor(props) {
+        super(props);
+        const extraOptions = this.getExtraOptions();
+        this.state.extraOptions = extraOptions;
+    }
+
+    setExtraOptions = (extraOptions) => {
+        this.setState({extraOptions});
```

Comment:
If we were to call that in the render method, it would cause a new object to be passed into AttachmentButton each time which would make it re-render. Since it's not a primitive object, I think it makes sense to put it in the state